students/alexLaws/lesson07/html_render.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+
+
+def style_tags(keyword_dict):
+    style = []
+    if keyword_dict is not None:
+        for key, value in keyword_dict.items():
+            style.append(' {}=\"{}\"'.format(key, value))
+    return "".join(style)
+
+
+class Element:
+    '''A class to render an HTML element'''
+
+    tag_name = ''
+    indents = 4
+
+    def __init__(self, content=None, **kwargs):
+        self.content = []
+        if content:
+            self.content.append(content)
+        self.kwargs = kwargs
+
+    def append(self, content):
+        '''Appends strings to content'''
+        self.content.append(content)
+
+    def render(self, file_out, cur_ind=""):
+        '''Render the HTML File'''
+
+        next_ind = cur_ind + (" " * self.indents)
+        element_format = style_tags(self.kwargs)
+        file_out.write("{}<{}{}>\n".format(cur_ind, self.tag_name, element_format))
+        for item in self.content:
+            if isinstance(item, str):
+                file_out.write('{}{}\n'.format(next_ind, item))
+            else:
+                item.render(file_out, next_ind)
+        file_out.write("{}</{}>\n".format(cur_ind, self.tag_name))
+
+
+class Html(Element):
+    '''A class for an element with an HTML tag'''
+
+    tag_name = 'html'
+
+    def __init__(self, content=None, **kwargs):
+        super().__init__(content, **kwargs)
+
+    def render(self, file_out, cur_ind=""):
+        '''Render HTML tags'''
+        file_out.write("{}<!DOCTYPE html>\n".format(cur_ind))
+        super().render(file_out, cur_ind)
+
+
+class Head(Element):
+    '''A class for an element with a head tag'''
+
+    tag_name = 'head'
+
+    def __init__(self, content=None, **kwargs):
+        super().__init__(content, **kwargs)
```

Comment:
if you don't have anything additional in the constructor you can omit `__init__` in the subclass and just have your class variable `tag_name`